[BUG]: repository_ruleset crash on empty required_deployment_environments

### Expected Behavior

Should define an empty list of required deployments. 

Equivalent of ticking the box in the UI but not specifying anything. See screenshot 
![Screenshot](https://github.com/integrations/terraform-provider-github/assets/66401219/d1683bf7-80b7-431a-abc0-2900cadf333d)


### Actual Behavior

Throws error
> The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain more details.

Seems like there was an error in [repository_rules_utils.go](https://github.com/integrations/terraform-provider-github/blob/69d1fcd12ab9294c06fc1aed23db47ffa9a4ee77/github/respository_rules_utils.go#L242).  It fails to cast the `required_deployments` block to a map of strings because `v[0] == nil`.  

### Terraform Version

Terraform v1.5.7

### Affected Resource(s)

- github_repository_ruleset

### Terraform Configuration Files

```hcl
resource "github_repository_ruleset" "failure" {
  repository  = <insert repo here>
  name        = "failure"
  enforcement = "evaluate"
  target      = "branch"

  rules {
    # just here to so that removing the required_deployments block is still valid
    update = true

    required_deployments {
      required_deployment_environments = []
    }
  }
}
```


### Steps to Reproduce

_No response_

### Debug Output

_No response_

### Panic Output

```shell
Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with
the following symbols:
  + create

Terraform will perform the following actions:

  # github_repository_ruleset.failure will be created
  + resource "github_repository_ruleset" "failure" {
      + enforcement = "evaluate"
      + etag        = (known after apply)
      + id          = (known after apply)
      + name        = "failure"
      + node_id     = (known after apply)
      + repository  = <censored>
      + ruleset_id  = (known after apply)
      + target      = "branch"

      + rules {
          + update                        = true
          + update_allows_fetch_and_merge = false

          + required_deployments {
              + required_deployment_environments = []
            }
        }
    }

Plan: 1 to add, 0 to change, 0 to destroy.
github_repository_ruleset.failure: Creating...
╷
│ Error: Request cancelled
│ 
│   with github_repository_ruleset.failure,
│   on main.tf line 26, in resource "github_repository_ruleset" "failure":
│   26: resource "github_repository_ruleset" "failure" {
│ 
│ The plugin.(*GRPCProvider).ApplyResourceChange request was cancelled.
╵

Stack trace from the terraform-provider-github_v5.37.0 plugin:

panic: interface conversion: interface {} is nil, not map[string]interface {}

goroutine 68 [running]:
github.com/integrations/terraform-provider-github/v5/github.expandRules({0xc0006196a0?, 0x1f5d882?, 0x1d69ac0?}, 0x0)
        github.com/integrations/terraform-provider-github/v5/github/respository_rules_utils.go:242 +0x2105
github.com/integrations/terraform-provider-github/v5/github.resourceGithubRulesetObject(0xc000c22960?, {0x0, 0x0})
        github.com/integrations/terraform-provider-github/v5/github/respository_rules_utils.go:32 +0x2c5
github.com/integrations/terraform-provider-github/v5/github.resourceGithubRepositoryRulesetCreate(0x1d69ac0?, {0x1cc4160?, 0xc000350400})
        github.com/integrations/terraform-provider-github/v5/github/resource_github_repository_ruleset.go:425 +0x56
github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Resource).Apply(0xc00016ab40, 0xc000c41950, 0xc000c22960, {0x1cc4160, 0xc000350400})
        github.com/hashicorp/terraform-plugin-sdk@v1.17.2/helper/schema/resource.go:320 +0x637
github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Provider).Apply(0xc00006ae00, 0xc0007eb900, 0x1f6909a?, 0xf?)
        github.com/hashicorp/terraform-plugin-sdk@v1.17.2/helper/schema/provider.go:294 +0x6a
github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin.(*GRPCProviderServer).ApplyResourceChange(0xc000802918, {0xc00096ee70?, 0x10bf886?}, 0xc00096ee70)
        github.com/hashicorp/terraform-plugin-sdk@v1.17.2/internal/helper/plugin/grpc_provider.go:895 +0x7c5
github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x1f0c0a0?, 0xc000802918}, {0x2325c10, 0xc00064e330}, 0xc00096ee00, 0x0)
        github.com/hashicorp/terraform-plugin-sdk@v1.17.2/internal/tfplugin5/tfplugin5.pb.go:3305 +0x169
google.golang.org/grpc.(*Server).processUnaryRPC(0xc00067a000, {0x232b3b0, 0xc000602680}, 0xc00064aea0, 0xc000573b90, 0x2c46ee0, 0x0)
        google.golang.org/grpc@v1.55.0/server.go:1337 +0xde7
google.golang.org/grpc.(*Server).handleStream(0xc00067a000, {0x232b3b0, 0xc000602680}, 0xc00064aea0, 0x0)
        google.golang.org/grpc@v1.55.0/server.go:1714 +0x9e7
google.golang.org/grpc.(*Server).serveStreams.func1.1()
        google.golang.org/grpc@v1.55.0/server.go:959 +0x8d
created by google.golang.org/grpc.(*Server).serveStreams.func1 in goroutine 31
        google.golang.org/grpc@v1.55.0/server.go:957 +0x165

Error: The terraform-provider-github_v5.37.0 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```


### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

--- a/github/resource_github_repository_ruleset.go
+++ b/github/resource_github_repository_ruleset.go
@@ -162,6 +162,7 @@ func resourceGithubRepositoryRuleset() *schema.Resource {
 									"required_deployment_environments": {
 										Type:        schema.TypeList,
 										Required:    true,
+										DefaultFunc: func() (interface{}, error) { return []string{}, nil },
 										Description: "The environments that must be successfully deployed to before branches can be merged.",
 										Elem: &schema.Schema{
 											Type: schema.TypeString,

--- a/website/docs/r/organization_ruleset.html.markdown
+++ b/website/docs/r/organization_ruleset.html.markdown
@@ -178,12 +178,18 @@ The `rules` block supports the following:
 
 #### bypass_actors ####
 
-* `actor_id` - (Required) (Number) The ID of the actor that can bypass a ruleset. When `actor_type` is `OrganizationAdmin`, this should be set to `1`.
+* `actor_id` - (Required) (Number) The ID of the actor that can bypass a ruleset.
 
 * `actor_type` (String) The type of actor that can bypass a ruleset. Can be one of: `RepositoryRole`, `Team`, `Integration`, `OrganizationAdmin`.
 
 * `bypass_mode` - (Optional) (String) When the specified actor can bypass the ruleset. pull_request means that an actor can only bypass rules on pull requests. Can be one of: `always`, `pull_request`.
 
+~>Note: at the time of writing this, the following actor types correspond to the following actor IDs:
+* `OrganizationAdmin` -> `1`
+* `RepositoryRole` (This is the actor type, the following are the base repository roles and their associated IDs.)
+  * `maintain` -> `2`
+  * `write` -> `4`
+  * `admiin` -> `5`
 
 #### conditions ####
 

--- a/website/docs/r/repository_ruleset.html.markdown
+++ b/website/docs/r/repository_ruleset.html.markdown
@@ -195,11 +195,18 @@ The `rules` block supports the following:
 
 #### bypass_actors ####
 
-* `actor_id` - (Required) (Number) The ID of the actor that can bypass a ruleset. When `actor_type` is `OrganizationAdmin`, this should be set to `1`.
+* `actor_id` - (Required) (Number) The ID of the actor that can bypass a ruleset.
 
 * `actor_type` (String) The type of actor that can bypass a ruleset. Can be one of: `RepositoryRole`, `Team`, `Integration`, `OrganizationAdmin`.
 
 * `bypass_mode` - (Optional) (String) When the specified actor can bypass the ruleset. pull_request means that an actor can only bypass rules on pull requests. Can be one of: `always`, `pull_request`.
+
+~> Note: at the time of writing this, the following actor types correspond to the following actor IDs:
+* `OrganizationAdmin` -> `1`
+* `RepositoryRole` (This is the actor type, the following are the base repository roles and their associated IDs.)
+  * `maintain` -> `2`
+  * `write` -> `4`
+  * `admiin` -> `5`
 
 
 #### conditions ####

